File: modules/kernel/examples/flags.py
## \example kernel/flags.py
# A simple example of how to use IMP flags support in a python
# executable. Run with "--help" to see all available flags.

import IMP
import sys

IMP.add_string_flag("hello", "Hello world", "How to say hello")

IMP.setup_from_argv(sys.argv, "An example for using flags in python.")

if IMP.get_string_flag("hello") == "":
    IMP.write_help()
    exit(1)

print(IMP.get_string_flag("hello"))

File: modules/kernel/examples/flags.cpp
/** \example flags.cpp
    Show simple usage of the command line flags support in IMP in a
    C++ executable. Run with "--help" to see all available flags.
*/
#include <IMP/flags.h>
#include <IMP/Flag.h>

int main(int argc, char *argv[]) {
  IMP::Flag<std::string> hello("hello", "How to say hello.",
                                     "Hello world");

  IMP::setup_from_argv(argc, argv, "Example on C++ use of flags.");

  std::cout << hello << std::endl;

  return 0;
}
